src/test/java/ibm/jceplus/junit/base/BaseTestECDHKeyAgreement.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright IBM Corp. 2024
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution.
+ */
+package ibm.jceplus.junit.base;
+
+import java.math.BigInteger;
+import java.security.KeyFactory;
+import java.security.KeyPair;
+import java.security.KeyPairGenerator;
+import java.security.interfaces.ECPrivateKey;
+import java.security.spec.ECPrivateKeySpec;
+import javax.crypto.KeyAgreement;
+import org.junit.jupiter.api.Test;
+
+public class BaseTestECDHKeyAgreement extends BaseTestJunit5 {
+
+    @Test
+    public void testInitWithInvalidKey() throws Exception {
+        KeyPairGenerator kpg = KeyPairGenerator.getInstance("EC", getProviderName());
+        kpg.initialize(256);
+        KeyPair kp = kpg.generateKeyPair();
+        ECPrivateKey privateKey = (ECPrivateKey) kp.getPrivate();
+
+        KeyFactory keyFactory = KeyFactory.getInstance("EC", getProviderName());
+        ECPrivateKey invalidPrivateKey
+                = (ECPrivateKey) keyFactory.generatePrivate(
+                        new ECPrivateKeySpec(BigInteger.ZERO,
+                                privateKey.getParams()));
+
+        KeyAgreement ka = KeyAgreement.getInstance("ECDH", getProviderName());
+
+        // The first initiation should succeed.
```

Comment:
I think that's `initialization`, not `initiation`. Same thing goes for the following comments.

Reply:
Updated